code/datums/status_effects/debuffs.dm

```diff
@@ -561,6 +561,84 @@
 		resist_debuff() // We repeat ourselves as long as the debuff persists.
 		return
 
+// ***************************************
+// *********** Melting fire
+// ***************************************
+/datum/status_effect/stacking/melting_fire
+	id = "melting_fire"
+	tick_interval = 2 SECONDS
+	stacks = 1
+	max_stacks = 5
+	consumed_on_threshold = FALSE
+	/// Owner of the debuff is limited to carbons.
+	var/mob/living/carbon/debuff_owner
+	/// Used for the fire effect.
+	var/obj/vis_melt_fire/visual_fire
+	var/decay_counter = 0
+
+/obj/vis_melt_fire
+	name = "ouch ouch ouch"
+	icon = 'icons/mob/OnFire.dmi'
+	layer = ABOVE_MOB_LAYER
+	vis_flags = VIS_INHERIT_DIR | VIS_INHERIT_ID | VIS_INHERIT_PLANE
+
+/datum/status_effect/stacking/melting_fire/on_creation(mob/living/new_owner, stacks_to_apply)
+	if(new_owner.status_flags & GODMODE)
+		qdel(src)
+		return
+	. = ..()
+	visual_fire = new(null)
+	visual_fire.icon_state = "melting_low_stacks"
+	debuff_owner = new_owner
+	debuff_owner.vis_contents += visual_fire
+	debuff_owner.balloon_alert(debuff_owner, "Melting fire")
+	playsound(debuff_owner.loc, "sound/bullets/acid_impact1.ogg", 30)
+	RegisterSignal(debuff_owner, COMSIG_LIVING_DO_RESIST, PROC_REF(call_resist_debuff))
+
+/datum/status_effect/stacking/melting_fire/on_remove()
+	debuff_owner.vis_contents -= visual_fire
+	QDEL_NULL(visual_fire)
+	debuff_owner = null
+	return ..()
+
+/datum/status_effect/stacking/melting_fire/tick()
+	. = ..()
+	if(!debuff_owner)
+		return
+	decay_counter++
+	if(decay_counter >= PYROGEN_STACK_FIRE_DECAY)
+		stacks--
+		decay_counter = 0
+	debuff_owner.take_overall_damage(PYROGEN_DAMAGE_PER_STACK * stacks, BURN, FIRE,updating_health = TRUE)
+	for(var/mob/living/carbon/xenomorph/pyrogen/fire_enjoyer in get_hearers_in_view(5, debuff_owner))
+		fire_enjoyer.gain_plasma(5, TRUE)
```

Comment:
this get hearers in view is gross. why is it not tracking the owner?

Reply:
kuro said in his doc to affect every pyrogen nearby.